### PR TITLE
feat: adapt Kafka bindings to v3

### PR DESCRIPTION
**Pull request by lbroudoux**
Signed-off-by: Laurent Broudoux <laurent.broudoux@gmail.com>

**Description**

As discussed and proposed [here](https://github.com/asyncapi/bindings/issues/182#issuecomment-1792837812) in issue #182, this PR: 
- Adapts README to explain with have different versions of bindings for v3 and v3,
- Add specific README to track v2 related changes and documentation,
- Add specific README to track v2 related changes and documentation.

Another PR with specific schemas for v2 and v3 is on its way on `asyncapi/spec-json-schemas`.

**Related issue(s)**

Relates to #182

### CODE REVIEW

File: kafka/README-v2.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+This document defines how to describe Kafka-specific information on AsyncAPI v2.x document.
+
+<a name="version"></a>
+
+## Version
+
+Current version is `v2-0.4.0` and is applicable to AsyncAPI v2.x only. For a version of Kafka binding applicable to AsyncAPI 3.x and examples on how to use it, please check this [README](./README-v3.md). 
+
+> `v2-0.4.0` is just a renmaing of previous `0.4.0` in order to better track future changes that could be only applicable into an AsyncAPI v2.x context.
```
Comment (dalelane) [inline kafka/README-v2.md:11]:
```suggestion
> `v2-0.4.0` is just a renaming of previous `0.4.0` in order to better track future changes that could be only applicable into an AsyncAPI v2.x context.
```

File: kafka/README-v2.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+<a name="channelBindingObjectReplicas"></a>`replicas` |                     integer                      |                              Number of replicas configured on this topic.                               |        OPTIONAL         | MUST be positive
+<a name="channelBindingObjectTopicConfiguration"></a>`topicConfiguration` | [TopicConfiguration Object](#topicConfiguration) |                   Topic configuration properties that are relevant for the API.                    |       OPTIONAL       | -
+<a name="channelBindingObjectBindingVersion"></a>`bindingVersion` |                      string                      |                   The version of this binding. If omitted, "latest" MUST be assumed.                    |   OPTIONAL [`latest`]   | -
+
+
+This object MUST contain only the properties defined above.
+
+##### Example
+
+This example is valid for any Confluent compatible schema registry. Here we describe the implementation using the first 4 bytes in payload to store schema identifier.
```
Comment (dalelane) [inline kafka/README-v2.md:61]:
```suggestion
```

File: kafka/README-v3.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+This document defines how to describe Kafka-specific information on AsyncAPI v3.x document.
+
+<a name="version"></a>
+
+## Version
+
+Current version is `v3-0.4.0` and is applicable to AsyncAPI v3.x only. For a version of Kafka binding applicable to AsyncAPI 2.x and examples on how to use it, please check this [README](./README-v2.md). 
+
+> `v3-0.4.0` is just a renmaing of previous `0.4.0` in order to better track future changes that could be only applicable into an AsyncAPI v3.x context.
```
Comment (dalelane) [inline kafka/README-v3.md:11]:
```suggestion
> `v3-0.4.0` is just a renaming of previous `0.4.0` in order to better track future changes that could be only applicable into an AsyncAPI v3.x context.
```

File: kafka/README-v3.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+<a name="channelBindingObjectReplicas"></a>`replicas` |                     integer                      |                              Number of replicas configured on this topic.                               |        OPTIONAL         | MUST be positive
+<a name="channelBindingObjectTopicConfiguration"></a>`topicConfiguration` | [TopicConfiguration Object](#topicConfiguration) |                   Topic configuration properties that are relevant for the API.                    |       OPTIONAL       | -
+<a name="channelBindingObjectBindingVersion"></a>`bindingVersion` |                      string                      |                   The version of this binding. If omitted, "latest" MUST be assumed.                    |   OPTIONAL [`latest`]   | -
+
+
+This object MUST contain only the properties defined above.
+
+##### Example
+
+This example is valid for any Confluent compatible schema registry. Here we describe the implementation using the first 4 bytes in payload to store schema identifier.
```
Comment (dalelane) [inline kafka/README-v3.md:61]:
```suggestion
```

**Review by dalelane — DISMISSED** [review]
spotted a few minor things we could fix, but overall this is exactly what we need

**Comment by lbroudoux** [comment]
All good: fix the typos and apply suggestions. Thanks @dalelane, for the review.

**Review by dalelane — DISMISSED** [review]

**Comment by lbroudoux** [comment]
Now replaced by #226.

Could you all confirm we call cancel this one?

**Comment by dalelane** [comment]
agreed

**Pull request closed without merging**